lib/puppet/type/k5login.rb

```diff
@@ -31,10 +32,59 @@
     defaultto { "644" }
   end
 
+  # To manage the selinux user of the file
+  newproperty(:seluser) do
+    desc "What the SELinux user component of the context of the file should be.
+      Any valid SELinux user component is accepted.  For example `user_u`.
+      If not specified it defaults to the value returned by matchpathcon for
+      the file, if any exists.  Only valid on systems with SELinux support
+      enabled."
+
+    defaultto { 'user_u' }
```

Comment:
I'm not sure if I'm missing context from previous iterations, but I'm curious about the default here. I know on the file type, it looks like it's getting the default context of the file in question.

 https://github.com/puppetlabs/puppet/blob/51636c065e0625b32af2cd98f2060eb7428704f5/lib/puppet/type/file/selcontext.rb#L103

Is there any reason you're not doing something similar? Or would that be part of the extra complication that @branan suggested staying away from?

Reply:
That makes sense